dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/VariantIterator.kt

```diff
@@ -0,0 +1,35 @@
+package com.datadog.gradle.plugin.internal
+
+/**
+ * This Iterator will take a list of flavor name and iterate over all possible
+ * (partial) variants from those.
+ *
+ * The partial variants will be iterated over in increasing order of priority.
+ * Product flavors that belong to the first flavor dimension have a higher priority than
+ * those belonging to the second flavor dimension.
+ *
+ * E.g.: for the variant with flavor names ["pro", "green", "europe"], the iterator will
+ * return the following values (in order):
+ * "europe", "green", "greenEurope", "pro", "proEurope", "proGreen", "proGreenEurope"
+ */
+internal class VariantIterator(
+    private val names: List<String>
+) : Iterator<String> {
+
+    private var index = 1
+    private val max = 1 shl names.size
+
+    override fun hasNext(): Boolean {
+        return index < max
+    }
+
+    override fun next(): String {
+        if (index >= max) {
+            throw NoSuchElementException()
```

Comment:
No because the underlying data cannot be modified. The contract for `Iterator` specifically says that if `next()` is called but `hasNext()` would return false, the implementation should throw a `NoSuchElementException`